### PR TITLE
create print-joblog and invoke it from support

### LINKED ISSUE #1107: tsocmd not working in zowe-support.sh
In zowe-support it tries to capture the output of the zowe job by calling tsocmd output, but this doesn’t seem to work on my system. When I call tsocmd "output ZWES2SV(STC63193)" it guess prints the command `output ZWES2SV(STC63193)` and even worse seems to purge the job from the system.
The doc page (https://www.ibm.com/support/knowledgecenter/sk/SSLTBW_2.3.0/com.ibm.zos.v2r3.bpxa500/tsonew.htm) states ’Do not issue 'tsocmd output' from the OMVS shell. If you do, you might not receive any output.' so maybe it is bound to fail, so does anyone have any suggestions of how else we can capture the job info into the support pax, or spot what we are doing wrong in zowe-support.sh?

We could consider using z/OSMF's API to get it, but I don’t think we can add that dependency to the zowe-support.sh, especially if z/OSMF, or configuration is the problem and it means we’d have to ship another script (either node, or Java) to make that possible

### PR DESCRIPTION
**Pull request by John-A-Davies**
Signed-off-by: John Davies <daviesja@uk.ibm.com>

Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [ ] Feature 
- [x] Other... Please describe: The documented restriction is that TSOCMD OUTPUT does not print the output if it's invoked from a USS shell script.  This fix issues it from a JCL job, not shell.

#### Relevant issues

Fixes #1107 

#### Changes proposed in this PR

- 
The TSO OUTPUT command is run by submitting a JCL job instead of using a shell `tsocmd output ...` instruction.  

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *
Gathering problem determination information.  
---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number: zowe/docs-site#1075

#### Other information

The site system programmer needs to ensure that the TSO OUTPUT command can collect Zowe STC jobs, and is not restricted to jobs starting with your userid.  This is true regardless of whether OUTPUT is run from a script or a job.  

See the [TSO/E installation exit IKJEFF53](https://www.ibm.com/support/knowledgecenter/SSLTBW_2.2.0/com.ibm.zos.v2r2.e0ze100/ikjeff53.htm) topic in IBM Knowledge Center.

### CODE REVIEW

File: bin/utils/print-joblog-to-file.sh [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    $SCRIPT zwe1sv  stc00406
+
+The output will be placed in $jobname-$jobid.log and the job will be purged.
+EndOfUsage
+echo script $SCRIPT ended 
+exit 
+fi
+
+jobname=$1
+jobid=$2
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: I think we want to consider passing in the jobname and id using opts as not doing that has caused problems elsewhere

File: bin/utils/print-joblog-to-file.sh [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+# Function: Print a JES job log to a USS file
+# Inputs  - jobname and jobid of job to be printed
+# Outputs - a USS file containing the job log, which will be placed in $jobname-$jobid.log
+#         - The job will be purged from the JES spool
+# Uses    - The TSO OUTPUT command
+
+# identify this script
+SCRIPT_DIR="$(dirname $0)"
+SCRIPT="$(basename $0)"
+echo script $SCRIPT started from $SCRIPT_DIR
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: Why is is echo needed? is it just debug?

File: bin/utils/print-joblog-to-file.sh [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+echo script $SCRIPT ended 
+exit 
+fi
+
+jobname=$1
+jobid=$2
+
+userid=${USER:-${USERNAME:-${LOGNAME}}}
+tsoJobname=RUNTSOCM
+tsoCommandOut=/tmp/print.job.log.$$.tso.out
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: We should use the `TMPDIR` variable if set as we've had issues in the past we people who have changed from `/tmp`. The rest of our code base uses `${TMPDIR:-/tmp}` to get around this, so we should be consistent here.

File: bin/utils/print-joblog-to-file.sh [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+cat > runtso1.jcl <<EndOfJCL1
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: Where will `runtso1.jcl` go? If we are called from the runtime/bin/utils folder then won't it go in there, which we treat as read-only? I think putting it in ${TMPDIR:-/tmp} is safer

File: bin/utils/print-joblog-to-file.sh [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+#          READY
+#          END
+
+# any extra message indicates an error
+
+if [[ `cat $tsoCommandOut | wc -l` -ne 4 ]]
+then
+    echo ====== TSO OUTPUT command log begin
+    cat $tsoCommandOut
+    echo ====== TSO OUTPUT command log end
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: this debug shouldn't go to stdout

File: bin/utils/print-joblog-to-file.sh [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    fi
+
+    echo script $SCRIPT failed
+    rm  $tsoCommandOut
+    exit 4
+fi
+
+rm  $tsoCommandOut
+
+cp "//$jobname.$jobid.outlist" $jobname-$jobid.log
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: Same as above - it looks like this log might end up written the the read-only runtime directory?

File: bin/utils/print-joblog-to-file.sh (hunk 1 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: Both parms are mandatory and there are no optional ones, so using opt would make the script and its usage more complicated.  This script is called from `zowe-support.sh` so we don't need to protect against manual errors.

File: bin/utils/print-joblog-to-file.sh (hunk 2 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: It's informational.

File: bin/utils/print-joblog-to-file.sh (hunk 3 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: Happy to use TMPDIR, but as the support script is likely to be needed when things have gone wrong, and outside of the install process, so TMPDIR probably won't be set.

File: bin/utils/print-joblog-to-file.sh (hunk 2 above)
Comment (stevenhorsman) [reply bin/utils/print-joblog-to-file.sh]: For who?

File: bin/utils/print-joblog-to-file.sh (hunk 3 above)
Comment (stevenhorsman) [reply bin/utils/print-joblog-to-file.sh]: TMPDIR is a system variable on unix, so I don't see what the install process has to do with it?

File: bin/utils/print-joblog-to-file.sh (hunk 4 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: Yes.

File: bin/utils/print-joblog-to-file.sh (hunk 5 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: This was intended to show the user the error message directly.  Happy to just append it to SUPPORT_ARCHIVE_LOG

File: bin/utils/print-joblog-to-file.sh (hunk 6 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: Yes, and that's where the original script would put them.  But now that we want the runtime to be read-only, it should go elsewhere.

File: bin/utils/print-joblog-to-file.sh (hunk 5 above)
Comment (stevenhorsman) [reply bin/utils/print-joblog-to-file.sh]: Ok, that's fine, I guess we'll need to make sure that the documentation mentions this

File: bin/utils/print-joblog-to-file.sh (hunk 2 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: It's for the person who invokes `zowe-support.sh`, to inform them that `print-joblog-to-file.sh` has started.  That invoker already sees many messages emitted, which are also appended to SUPPORT_ARCHIVE_LOG.  To be fair,  `print-joblog-to-file.sh` is the only script that  `zowe-support.sh` calls, but I'm happy to adopt Zowe house style.

File: bin/utils/print-joblog-to-file.sh (hunk 3 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: OK, will use TMPDIR.

File: bin/utils/print-joblog-to-file.sh (hunk 6 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: It now goes where `zowe-support.sh` tells it to go.

File: bin/utils/print-joblog-to-file.sh [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+EndOfUsage
+echo script $SCRIPT ended 
+exit 
+fi
+
+jobname=$1
+jobid=$2
+jobfile=$3
+
+userid=${USER:-${USERNAME:-${LOGNAME}}}
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: What is this line of code for?

File: bin/utils/print-joblog-to-file.sh [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+tsoJobname=RUNTSOCM
+if [ -z ${TEMP_DIR+x} ]; then
```
Comment (stevenhorsman) [inline bin/utils/print-joblog-to-file.sh]: What is the ${TEMP_DIR+x} supposed to do? It just seems to evaluate to "x" when I try it?

Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: I've removed the IF at line 53.

File: bin/utils/print-joblog-to-file.sh (hunk 7 above)
Comment (John-A-Davies) [reply bin/utils/print-joblog-to-file.sh]: Left over, removed.

File: bin/zowe-support.sh [hunk 9]
```diff
@@ -88,16 +88,22 @@ add_to_pax $PS_OUTPUT_FILE process_info
 rm $PS_OUTPUT_FILE
 
 # Collect STC output
-STC_JOBS=`tsocmd "STATUS ${ZOWE_STC}" 2>/dev/null | grep -E 'OUTPUT' | cut -d' ' -f2`
-for STC in ${STC_JOBS}
+if [ -z ${TEMP_DIR+x} ]; then
```
Comment (stevenhorsman) [inline bin/zowe-support.sh]: I think this if should be removed. TEMP_DIR doesn't exist at this point

Comment (John-A-Davies) [reply bin/zowe-support.sh]: Done.

**Comment by John-A-Davies** [comment]
Corresponding doc update:
https://github.com/zowe/docs-site/pull/1075

**Comment by John-A-Davies** [comment]
I added a link to the doc update issue to the PR description

File: bin/zowe-support.sh (hunk 9 above)
Comment (stevenhorsman) [reply bin/zowe-support.sh]: @John-A-Davies - when is the commit that you've done this in going to be pushed? I think it's all that is holding the merge up

Comment (John-A-Davies) [reply bin/zowe-support.sh]: I think I removed it this time.

**Comment by John-A-Davies** [comment]
Closing in order to perform DCO signoff check in another PR

**Pull request closed without merging**